Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -65,6 +68,9 @@ - (id)initWithDrawingAreaProxy:(WebKit::RemoteLayerTreeDrawingAreaProxy*)drawing
     if (self = [super init]) {
         _drawingAreaProxy = drawingAreaProxy;
         // Note that CADisplayLink retains its target (self), so a call to -invalidate is needed on teardown.
+#if ENABLE(TIMER_DRIVEN_DISPLAY_REFRESH_FOR_TESTING)
+        if (![NSUserDefaults.standardUserDefaults integerForKey:@"MainScreenRefreshRate"])
```

Comment:
Don't hide flow of control inside #idefs. Give the class a `std::optional<FramesPerSecond> m_overrideFrameRate` data member. Then here have:

```
#if ENABLE(TIMER_DRIVEN_DISPLAY_REFRESH_FOR_TESTING)
  NSNumber* overrideRefreshRate = [NSUserDefaults.standardUserDefaults integerForKey:@"MainScreenRefreshRate"];
  NSInteger overrideRefreshRateValue = [overrideRefreshRate integerValue];
  if (overrideRefreshRateValue)
    m_overrideFrameRate = overrideRefreshRateValue;
```

Then you can test `m_overrideFrameRate` in places where you decide to make the display link or timer.